benchmarks/filterapi.h

```diff
@@ -27,6 +27,11 @@
 #include "vqf_cpp.h"
 #include "simd-block.h"
 #endif
+#define __PF_AVX512__  (__AVX512BW__ & __AVX512VL__ & __AVX512CD__ & __AVX512DQ__)
+#ifdef __PF_AVX512__
```

Comment:
```suggestion
#if __PF_AVX512__
```

Reply:
The `#ifdef __PF_AVX512__` will always be true because you define the macro on the previous line.

Reply:
Makes sense! An alternative might be, similar to https://stackoverflow.com/questions/39290019/is-if-defined-macro-equivalent-to-ifdef-macro -- but I didn't test:

```
#if defined (__AVX512BW__) && defined (__AVX512VL__) && ...
#define __PF_AVX512__
#endif

#ifdef __PF_AVX512__
```